### DISCUSSION TITLE
Not able to add key value in input JSON inside vert.x gateway.

### DISCUSSION THREAD

**Discussion by iamakashk** [apiman/apiman#1412] [Help]
Hi I am using apiman vert.x 1.5.7 and  trying to add key in my input request JSON before sending it to backend but somehow it is not changing/ the key is not getting added 
```

		// Write data into executor. Called when ready.
		        executor.streamHandler(writeStream -> {
            vertxRequest.handler(bufferChunk -> {
            	
            	bufferChunk.writeToBuffer(bufferChunk.toJsonObject().put("randomkey","randomkey").toBuffer());

                // Apply back-pressure.
                if (writeStream.isFull()) {
                    vertxRequest.pause();
                }
                // Wrap Vert.x buffer into apiman's buffer IR, then write into engine.
                writeStream.write(new VertxApimanBuffer(bufferChunk));
            });

            // Called when back-pressure has reduced sufficiently to resume.
            writeStream.drainHandler(onDrain -> vertxRequest.resume());

            vertxRequest.endHandler(end -> writeStream.end());

            // Now safe to resume body transmission.
            vertxRequest.resume();
        });

```

am i doing something wrong here ?

**Comment by msavy** [comment]
Are you writing a custom policy? 

If so, you are not supposed to be using Vert.x directly. Instead you should be using Apiman's custom policy framework. What you want to do should be relatively easy using the functions that already exist. Check out the transformation policy if you want an example.

BTW, if you want to play with data you need to be implementing the IDataPolicy interface.

Also, you can refer to the docs on this topic -- https://www.apiman.io/apiman-docs/core/2.1.0.Final/development/Guide.html#_idata_policy
